lib/src/Pos/Launcher/Runner.hs

```diff
@@ -111,19 +114,20 @@ elimRealMode NodeResources {..} action = do
 -- | "Batteries-included" server.
 -- Bring up a full diffusion layer over a TCP transport and use it to run some
 -- action. Also brings up ekg monitoring, route53 health check, statds,
--- according to parameters. 
+-- according to parameters.
 runServer
     :: forall ctx m t .
        ( DiffusionWorkMode m
        , LogicWorkMode ctx m
+       , HasShutdownContext ctx
        , MonadFix m
        )
     => NodeParams
     -> EkgNodeMetrics m
     -> OutSpecs
     -> ActionSpec m t
     -> m t
-runServer NodeParams {..} ekgNodeMetrics _ (ActionSpec act) =
+runServer NodeParams {..} ekgNodeMetrics _ (ActionSpec act) = const (exitWith (ExitFailure 20)) =<< (race waitForShutdown $
```

Comment:
I think it would be a bit more understandable (and the line would be not so big) if you moved this logic into a separate function in `where` clause (e. g. `exitOnShutdown action = const (exitWith (ExitFailure 20)) =<< (race waitForShutdown action)`).